modules/javafx.controls/src/main/java/javafx/scene/control/input/BehaviorBase2.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javafx.scene.control.input;
+
+import java.util.Objects;
+import javafx.event.Event;
+import javafx.event.EventHandler;
+import javafx.event.EventType;
+import javafx.scene.control.Control;
+import javafx.scene.control.Skin;
+import javafx.scene.input.KeyCode;
+
+/**
+ * Class provides a foundation for behaviors.
+ * <p>
+ * A concrete behavior implementation should do three things:
+ * 1. provide default behavior methods (a.k.a. functions)
+ * 2. in install() method, called from Skin.install(), map control's function tags to
+ *    behavior methods, map key bindings to function tags, and add additional event handlers,
+ *    using func(), key(), and hand() methods correspondingly.
+ *    Important: no mapping should be made in the behavior constructor, only in install().
+ * <p>
+ * The base class adds a dispose() method (called from Skin.dispose()),
+ * which undoes the mapping done in install().
+ * <p>
+ * TODO rename BehaviorBase and remove IBehavior
+ */
+public abstract class BehaviorBase2<C extends Control> implements IBehavior {
```

Comment:
I think Control is the best candidate to have an InputMap.  There is currently no reason to have it in Node, since all the instantiations of the existing InputMap are done with Controls.  Same applies to (current) Behaviors.